### ISSUE TITLE
it.otherNativeCrash must not be null

### ISSUE THREAD

**Issue by RainNapper** [Flank/flank#1026]
Running on 20.08.2

This seems to impact ~1/5 of my test runs. It could be the test runs with crashes, manually looking through some of the matrices.

```
java.lang.IllegalStateException: it.otherNativeCrash must not be null
        at ftl.json.OutcomeDetailsFormatterKt.getFailureOutcomeDetails(OutcomeDetailsFormatter.kt:43)
        at ftl.json.OutcomeDetailsFormatterKt.getDetails(OutcomeDetailsFormatter.kt:19)
        at ftl.reports.outcome.TestOutcomeKt.createMatrixOutcomeSummaryUsingEnvironments$default(TestOutcome.kt:18)
        at
ftl.reports.outcome.CreateMatrixOutcomeSummaryKt.createMatrixOutcomeSummary(CreateMatrixOutcomeSummary.kt:8)
        at ftl.json.SavedMatrix.updateFinishedMatrixData(SavedMatrix.kt:94)
        at ftl.json.SavedMatrix.finished(SavedMatrix.kt:90)
        at ftl.json.SavedMatrix.updateState(SavedMatrix.kt:74)
        at ftl.json.SavedMatrix.update(SavedMatrix.kt:61)
        at ftl.json.MatrixMapKt.update(MatrixMap.kt:57)
        at ftl.run.NewTestRunKt$newTestRun$2$2.invokeSuspend(NewTestRun.kt:24)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:47)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
        at picocli.CommandLine.access$1100(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
        at picocli.CommandLine.execute(CommandLine.java:2058)
        at ftl.Main$Companion$main$1.invoke(Main.kt:59)
        at ftl.Main$Companion$main$1.invoke(Main.kt:49)
        at ftl.util.Utils.withGlobalExceptionHandling(Utils.kt:130)
        at ftl.Main$Companion.main(Main.kt:55)
        at ftl.Main.main(Main.kt)
Exception occurred running test retries
org.gradle.process.internal.ExecException: Process 'command '/usr/lib/jvm/openjdk8/bin/java'' finished with non-zero exit value 1
```

**Comment by pawelpasterz** [comment]
The error started to appear after version change `2.07.0` to `2.08.02`

**Comment by RainNapper** [comment]
On my CI, I dropped from 2.08.02 to 2.08.01 and have not seen this issue in ~100 runs overnight. Hopefully that helps to narrow down the culprit.

**Comment by RainNapper** [comment]
Config used in the runs that crashed:
```> Task :Pinterest:flankTestTablet
Tests Filter Specified as 'package com.pinterest.tablet'
Testing with the following filters: package com.pinterest.tablet
version: v20.08.2
revision: 5f9a81d148ae89a84e5c12410eef75ea9abbf698
 
AndroidArgs
    gcloud:
      results-bucket: test-lab-vw4y21tcm9700-i1dn5xsbfdmaq
      results-dir: 2020-08-19_23-10-54.465000_GGGA
      record-video: true
      timeout: 5m
      async: false
      client-details:
      network-profile: null
      results-history-name: null
      # Android gcloud
      app: /project/Pinterest/build/artifacts/Pinterest/apk/Pinterest-ota-debug.apk
      test: /project/Pinterest/build/artifacts/Pinterest/apk/Pinterest-ota-debug-androidTest.apk
      additional-apks:
      auto-google-login: false
      use-orchestrator: true
      directories-to-pull:
      other-files:
      performance-metrics: false
      num-uniform-shards: null
      test-runner-class: null
      test-targets:
        - package com.pinterest.tablet
      robo-directives:
      robo-script: null
      device:
        - model: Nexus9
          version: 25
          locale: en
          orientation: portrait
      num-flaky-test-attempts: 0
 
    flank:
      max-test-shards: 250
      shard-time: 120
      num-test-runs: 1
      smart-flank-gcs-path: gs://test-lab-test-project.appspot.com/flank/smart-flank-cache/1424704448.xml
      smart-flank-disable-upload: false
      files-to-download:
      test-targets-always-run:
      disable-sharding: false
      project: test-lab-test-project
      local-result-dir: /project/Pinterest/build/flank/results
      full-junit-result: false
      # Android Flank Yml
      keep-file-path: false
      additional-app-test-apks:
      run-timeout: -1
      legacy-junit-result: false
      ignore-failed-tests: false
      output-style: multi
 ```

**Comment by bootstraponline** [comment]
@RainNapper this should be fixed in Flank v20.08.3
